modules/shapefile/src/lib/streaming/zip-batch-iterators.js

```diff
@@ -36,21 +37,50 @@ export async function* zipBatchIterators(iterator1, iterator2) {
 /**
  * Extract batch of same length from two batches
  *
- * @param  {Array} batch1
- * @param  {Array} batch2
- * @return {Array?}
+ * @param  {array | object} batch1
```

Comment:
Do we need to support both variants? Or shall we just standardize batches?

Reply:
I originally thought it would be preferable to include existing Array support for backwards compatibility, but I suppose this is not exported and used only by `the ShapefileLoader`.